internal/execext/exec.go

```diff
@@ -36,6 +39,31 @@ func RunCommand(ctx context.Context, opts *RunCommandOptions) error {
 		return ErrNilOptions
 	}
 
+	// Set "-e" or "errexit" by default
+	opts.POSIXOpts = append(opts.POSIXOpts, "e")
+
+	// Format POSIX options into a slice that mvdan/sh understands
+	var params []string
+	for _, opt := range opts.POSIXOpts {
+		if len(opt) == 1 {
+			params = append(params, fmt.Sprintf("-%s", opt))
+		} else {
+			params = append(params, "-o")
+			params = append(params, opt)
+		}
+	}
+
+	var s []string
+
+	// Manually adding bash opts before each command as there is no interp.Shopts
+	if len(opts.BashOpts) > 0 {
+		s = append(s, fmt.Sprintf("shopt -s %s", strings.Join(opts.BashOpts, " ")))
+	}
+
+	if len(s) > 0 {
+		opts.Command = strings.Join(append(s, opts.Command), " && ")
```

Comment:
Adding `interp.BashOpts` would be better, but if we need to do this ourselves, I would do it differently.

We can call `r.Run` multiple times, so it's just a matter of calling every `shopt` command before the user command. This would be safer than changing `opts.Command`.